maps-compose/src/main/java/com/google/maps/android/compose/GoogleMap.kt

```diff
@@ -95,11 +96,25 @@ public fun GoogleMap(
         return
     }
 
-    val context = LocalContext.current
-    val mapView = remember { MapView(context, googleMapOptionsFactory()) }
+    // Will either be set to a re-used or a new MapView
+    var mapViewOrNull: MapView? by remember { mutableStateOf(null) }
+    var isMapViewReused by remember { mutableStateOf(true) }
+
+    AndroidView(
+        modifier = modifier,
+        factory = { context ->
+            isMapViewReused = false
+            MapView(context, googleMapOptionsFactory())
+        },
+        onReset = { },
+        onRelease = { it.destroyAndRemoveAllViews() },
+        update = { mapViewOrNull = it }
+    )
+
+    // Wait until we have a MapView
+    val mapView = mapViewOrNull ?: return
```

Comment:
Thank you for your feedback and good observations. I am working on fixing these issues.

You are right that waiting for the MapView to be present before initializing it can result in a slight slowdown. Therefore I think it could be a good idea to allow the user to optionally enable reuse of the underlying MapView using a parameter with a default value of false.

Reply:
Thanks. There may be other ways to structure `AndroidView()` + lifecycle observing. Perhaps a refactor in this area could help avoid adding a flag.

Reply:
I will try

Reply:
I did some tests to find any performance implications with waiting for a reference to MapView before applying map initializations/configurations and these are the results. Note that this is only initial composition/map creation. I restarted the app with a single map each time.

With reusable AndroidView implementation
Average: 539.46 ms
Median: 462 ms

With current implementation:
Average: 538.56 ms
Median: 451 ms

As you can see there is virtually no difference. If I did many more samples I'm sure the median would be even closer.

But as you stated in your other comment it's possible we don't need to bother about this if we integrate the code below to the early return into the AndroidView.

Reply:
`AndroidView()` `factory` and `update` are called after initial composition during the Composer "apply" phase. The delay would depend on the time needed for initial composition, might be fairly quick or fairly long. There are other unnecessary delays in the initialization chain, which I noted in a couple issues recently. It would not be great to add more delay potential if it's avoidable.